configurator/configurator.go

```diff
@@ -28,205 +31,301 @@ import (
-	resolver trireme.PolicyResolver,
-	processor enforcer.PacketProcessor,
-	eventCollector collector.EventCollector,
-	secrets secrets.Secrets,
-	impl constants.ImplementationType) trireme.Trireme {
+// TriremeResult is the result of the creation of Trireme
+type TriremeResult struct {
+	Trireme        trireme.Trireme
+	DockerMonitor  monitor.Monitor
+	RPCMonitor     rpcmonitor.RPCMonitor
+	PublicKeyAdder enforcer.PublicKeyAdder
+	Secret         secrets.Secrets
+}
 
-	if eventCollector == nil {
-		zap.L().Warn("Using a default collector for events")
-		eventCollector = &collector.DefaultCollector{}
-	}
+// DefaultTriremeOptions returns a default set of options.
+func DefaultTriremeOptions() *TriremeOptions {
+	return &TriremeOptions{
+		TargetNetworks: []string{},
 
-	rpcwrapper := rpcwrapper.NewRPCWrapper()
+		EventCollector: &collector.DefaultCollector{},
 
-	enforcers := map[constants.PUType]enforcer.PolicyEnforcer{
-		constants.ContainerPU: enforcerproxy.NewDefaultProxyEnforcer(
-			serverID,
-			eventCollector,
-			secrets,
-			rpcwrapper,
-			DefaultProcMountPoint,
-		),
-	}
+		DockerSocketType: constants.DefaultDockerSocketType,
+		DockerSocket:     constants.DefaultDockerSocket,
 
-	s, err := supervisorproxy.NewProxySupervisor(eventCollector, enforcers[0], rpcwrapper)
+		Validity: time.Hour * 8760,
 
-	if err != nil {
-		zap.L().Fatal("Cannot initialize proxy supervisor", zap.Error(err))
-	}
+		FilterQueue:             fqconfig.NewFilterQueueWithDefaults(),
+		ExternalIPCacheValidity: -1, // Will get the default from the instantiation.
 
-	supervisors := map[constants.PUType]supervisor.Supervisor{constants.ContainerPU: s}
-	return trireme.NewTrireme(serverID, resolver, supervisors, enforcers, eventCollector)
+		ModeType: constants.RemoteContainer,
+		ImplType: constants.IPTables,
+
+		ProcMountPoint:        DefaultProcMountPoint,
+		AporetoProcMountPoint: DefaultAporetoProcMountPoint,
+
+		RemoteArg: constants.DefaultRemoteArg,
+
+		RPCAddress:              rpcmonitor.DefaultRPCAddress,
+		LinuxProcessReleasePath: "",
+
+		MutualAuth: false,
+
+		KillContainerError: false,
+		SyncAtStart:        true,
+
+		PKI: false,
+
+		LocalProcess:    true,
+		LocalContainer:  false,
+		RemoteContainer: true,
+		CNI:             false,
+	}
 }
 
-// NewHybridTrireme instantiates Trireme with both Linux and Docker enforcers.
-// The Docker enforcers are remote
-func NewHybridTrireme(
-	serverID string,
-	resolver trireme.PolicyResolver,
-	processor enforcer.PacketProcessor,
-	eventCollector collector.EventCollector,
-	secrets secrets.Secrets,
-	networks []string,
-) trireme.Trireme {
+// NewTriremeWithOptions creates all the Trireme objects based on the option struct
+func NewTriremeWithOptions(options *TriremeOptions) (*TriremeResult, error) {
 
-	if eventCollector == nil {
-		zap.L().Warn("Using a default collector for events")
-		eventCollector = &collector.DefaultCollector{}
+	enforcers := map[constants.PUType]enforcer.PolicyEnforcer{}
```

Comment:
Can you add some basic checks here. For example:
1. At least one of the options must be there (container or linux)
2. Local containers and process not supported together.